packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -4369,67 +4369,343 @@ class PositionedDirectional extends StatelessWidget {
+///
+/// [Flexible] : A widget that _could_ cause its child to stretch in the
+/// direction of the main axis, _though it does necessarily have to._
+///
+/// [Expanded] : A type of [Flexible] that _must_ stretch its child
+/// to use all of the space _reserved for it_ during layout.
+///
+/// It is important to understand an [Expanded] will always use all of the
+/// space it is offered, _but it may not be offered all of the leftover
+/// space there is._ There is an edge case in which a [Flexible] and an
+/// [Expanded] are used within the same [Flex], and the results can only
+/// be understood once a person understands what is happening within this
+/// layout algorithm. More on that later.
+///
+/// [Flexible.fit] : The fit is the parameter of the [Flexible] that
+/// determines if it must stretch its children.
+///
+/// [FlexFit] : An enum that is used by the [Flexible.fit]. There are
+/// two values, tight and loose:
+///
+/// [FlexFit.tight] will force the child to stretch and use all of
+/// the space offered to it _in the main axis_. An [Expanded] is
+/// nothing more than a [Flexible] with its fit parameter hard
+/// coded to [FlexFit.tight]. It is important to remember that
+/// any size specified in the child of an [Expanded] will be ignored,
+/// as the child will be forced to use all of the space offered to it
+/// in the main axis.
+///
+/// Any time a widget is given "tight constraints" by its parent,
+/// that widget will be forced to use all of the space offered to it.
+///
+/// [FlexFit.loose] is the default fit for the [Flexible]. It will not cause
+/// its child to stretch in the main axis, and will allow the child to be the
+/// size the child's parameters state it should be.
+///
+/// [MainAxisSize] : An enum that that determines if the [Flex] will
+/// be forced to be as large as its parent will allow.
+///
+/// [MainAxisSize.max] will force the [Flex] to be as large as the parent
+/// will allow in the main axis. _This does not force the children to be
+/// larger._
+///
+/// [MainAxisSize.min] : Allow the [Flex] to be whatever size it needs
+/// to be in order to accommodate its children, without forcing the
+/// [Flex] to be any larger than the total size of all its children.
+///
+/// [MainAxisAlignment] : How to align, center, or apply spacing
+/// between the children in the direction in the main axis.
+///
+/// [CrossAxisAlignment] : How to align, or stretch, the children in
+/// the direction of the cross axis.
+///
+/// Constraints: These are the minimum and maximum sizes between
+/// which a widget must be. There are constraints both vertically and
+/// horizontally. If constraints are _tight_, this means the min and max
+/// limits are the same, forcing the widget to be that specific size. If
+/// constraints are loose, the widget can be any size within the specified
+/// min and max, for both width and height.
+///
+/// There is another case in which there are no constraints given,
```

Comment:
Well, there are constraints given, but some or all parameters are set to `double.infinity` or zero.